modules/developer_manual/pages/webdav_api/trashbin.adoc

```diff
@@ -0,0 +1,229 @@
+Specifically, it provides the following functionality:
+
+* xref:list-files[List files in the trash bin]
+* xref:delete-files[Delete files permanently from the trash bin]
+* xref:restore-deleted-file[Restore deleted files to the trash bin]
+
+== List Files
+
+List all files in the trash bin
+
+* Path: `remote.php/dav/trash-bin/<username>`
+* Method: `PROPFIND`
+
+=== Request Parameters
+
+[cols=",,",options="header",]
+|===
+|Attribute 
+|Type 
+|Description
+
+|`username` 
+|string 
+|The username of the user to list files for.
+|===
+
+=== Request Headers
+
+[cols=",,",options="header",]
+|===
+|Header
+|Type
+|Description
+
+|`Depth`
+|int
+a|The maximum level to descend to, when retrieving a listing of the files in the user's trash bin.
+
+NOTE: Only one level of files and directories is returned if this header is omitted.
+|===
+
+=== Code Example
+
+[tabs]
+====
+Curl::
++
+--
+[source,console]
+....
+include::{examplesdir}core/scripts/curl/dav/trashbin_api/list-files-in-trashbin.sh[]
+....
+--
+====
+
+=== Returns
+
+This method returns an HTTP 207 (Multi-Status) status code and an XML response that contains the files in the trash bin of the authenticating user, limited to the depth specified by the `Depth` header.
+
+==== Example Response
+
+If the user that you’re connecting with is authorized, then you will see output similar to the following:
+
+[tabs]
+====
+Curl::
++
+--
+[source,xml]
+....
+include::{examplesdir}core/webdav_api/trashbin/list-files-in-trashbin-success-response.xml[]
+....
+--
+====
+
+== Delete Files
+
+Permanently delete a file from the trash bin.
+
+* Path: `remote.php/dav/trash-bin/<username>/<fileid>`
+* Method: `DELETE`
+
+=== Request Parameters
+
+[cols=",,",options="header",]
+|===
+|Attribute 
+|Type 
+|Description
+
+|`username` 
+|string 
+|The username of the user to list files for.
+
+|`fileid` 
+|integer 
+|The id of the file to delete.
+|===
+
+=== Code Example
+
+[tabs]
+====
+Curl::
++
+--
+[source,console]
+....
+include::{examplesdir}core/scripts/curl/dav/trashbin_api/delete-file-from-trashbin.sh[]
+....
+--
+====
+
+=== Returns
+
+This method returns an HTTP 201 (Created) status code, if the file is successfully restored.
+If there is no file in the trash bin with the specified file id, then an HTTP 404 (Not Found) status code is returned.
+
+== Restore Deleted File
+
+Restore a file from the trash bin.
+
+* Path: `remote.php/dav/trash-bin/<username>/<fileid>`
+* Method: `MOVE`
+
+=== Request Parameters
+
+[cols=",,",options="header",]
+|===
+|Attribute 
+|Type 
+|Description
+
+|`username` 
+|string 
+|The username of the user to list files for.
+
+|`fileid` 
+|integer 
+|The id of the file to delete.
+|===
+
+=== Request Headers
+
+[cols=",,,",options="header",]
+|===
+|Header
+|Type
+|Value
+|Description
+
+|`Overwrite`
+|string
+|`F` or `T`
+|Whether to overwrite an existing file.
+
+|`Destination`
+|string
+|`${SERVER_URI}/remote.php/dav/files/<username>/<fileid>`
```

Comment:
fileid is wrong here - needs to be the full path - please add a note that the parent folder of the destination has to exist

Reply:
The need for the destination directory to exists is covered at the end of the "Returns" section for this endpoint.